### PR TITLE
Changes to book loader

### PR DESCRIPTION
**Pull request by mhasself**
Book -> AxisManager loader module is substantially rewritten.

Now:
- `load_book_file()` can be used to load any .g3 file from an obs/oper book, without obsfiledb support.  In fact you can load (and concatenate) multiple files from any single detset.  The output is basically identical to what you'd get loading them through obsfiledb/Context.  This addresses #514 .
- The Book "ancil" data (i.e. position encoders) are loaded into the AxisManager.  In the present design, they are all simply copied into a sub-axisman called "ancil".  They are also copied, and scaled, into a sub-axisman called "boresight".  As the data are processed I am imagining that "ancil" will remain untouched but "boresight" might be updated (or recomputed) based on a pointing model. 
- Wiring frames are parsed, using SmurfStatus, and key info include in .det_info.smurf and .iir_params.<stream_id>.

Here's what loaded TOD looks like:
```
>>> tod
AxisManager(timestamps[samps], ancil*[samps], boresight*[samps], signal[dets,samps],
  biases[bias_lines,samps], primary*[samps], iir_params*, det_info*[dets], flags*[dets,samps],
  dets:LabelAxis(10336), samps:OffsetAxis(100000), bias_lines:LabelAxis(72))
>>> tod.det_info
AxisManager(detset[dets], stream_id[dets], smurf*[dets], dets:LabelAxis(10336))
>>> tod.det_info.smurf
AxisManager(band[dets], channel[dets], frequency[dets], rchannel[dets], dets:LabelAxis(10336))
>>> tod.iir_params
AxisManager(per_stream, ufm_mv22*, ufm_mv19*, ufm_mv6*, ufm_mv18*, ufm_mv14*, ufm_mv9*)
>>> tod.iir_params.ufm_mv6
AxisManager(enabled, b[16], a[16], fscale)
```

### CODE REVIEW

File: sotodlib/io/load_book.py [hunk 1]
```diff
@@ -145,170 +96,603 @@ def load_obs_book(db, obs_id, dets=None, prefix=None, samples=None,
...
+    if stat.num_chans is None:
+        # No wiring frames probably means it's an A_* (ancil) file.
+        stat = None
+    else:
+        # This stuff will need to be regrouped, for all detectors.  So
+        # just extract individual items, restricting to dets actually
+        # used here.
+        ch_info_full = load_smurf.get_channel_info(stat)
+        ch_info = {}
+        for k, v in ch_info_full._fields.items():
```
Comment (kmharrington) [inline sotodlib/io/load_book.py:277]: `get_channel_info` accepts a mask you can use to already down-select the channel information.

**Review by kmharrington — CHANGES_REQUESTED** [review]
* When loading "/so/staged/satp1/obs/16918/obs_1691876375_satp1_0111000/D_ufm_mv18_000.g3"
`aman.boresight.roll` is None? There shouldn't be any mount info in that file. 

* Trying to load a file that does not include the first file in on observation throws an un-informative error. Reason is the status object might exist but is incomplete. In `load_smurf.load_file` [we just blindly try to use the first observation file before throwing an error](https://github.com/simonsobs/sotodlib/blob/6cde84ce9dad2e4f927841691a4f539da735f947/sotodlib/io/load_smurf.py#L2839). 
```
>>> aman = lb.load_book_file("/so/staged/satp1/obs/16918/obs_1691876375_satp1_0111000/D_ufm_mv18_001.g3")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/kmharrin/.local/lib/python3.10/site-packages/sotodlib/io/load_book.py", line 169, in load_book_file
    return _concat_filesets({'?': this_detset},
  File "/home/kmharrin/.local/lib/python3.10/site-packages/sotodlib/io/load_book.py", line 388, in _concat_filesets
    for k, v in r['iir_params'].items():
AttributeError: 'NoneType' object has no attribute 'items'
```
* `load_book_files` can't mix files from different stream ids. Ex: 
```
["/so/staged/satp1/obs/16918/obs_1691876375_satp1_0111000/D_ufm_mv18_000.g3",
 "/so/staged/satp1/obs/16918/obs_1691876375_satp1_0111000/D_ufm_mv22_000.g3", 
 "/so/staged/satp1/obs/16918/obs_1691876375_satp1_0111000/D_ufm_mv29_000.g3"]
``` 
At a minimum I think that should be in the docstring.

**Comment by mhasself** [comment]
Thanks for the review -- I believe all your comments have been addressed.

"boresight" is now not created at all unless az and el are both found in ancil.

Also I added a wildcard thing you can just do this to get a whole stream_id:
```
>>> aman = lb.load_book_file("/so/staged/satp1/obs/16918/obs_1691876375_satp1_0111000/D_ufm_mv18_*.g3")
```

**Review by kmharrington — APPROVED** [review]
Yup looks good. Thanks!

**Comment by msilvafe** [comment]
> Also I added a wildcard thing you can just do this to get a whole stream_id:
> 
> ```
> 
> >>> aman = lb.load_book_file("/so/staged/satp1/obs/16918/obs_1691876375_satp1_0111000/D_ufm_mv18_*.g3")
> 
> ```

Love this

**Pull request merged**